pipelines/nextflow/subworkflows/genome_prepare/genome_prepare.nf

```diff
@@ -62,50 +40,67 @@ include { MANIFEST_STATS } from '../../modules/manifest_stats.nf'
 workflow GENOME_PREPARE {
 
     take:
-        accession
-        genome_json
-        output_dir
+        genomic_dataset // tuple composed of GCA_XXXXXXX.X (as path) and genome.json
+        output_dir // User specified or default
         cache_dir
 
-    main:
-        CHECK_JSON_SCHEMA_GENOME(genome_json, accession)
-        DOWNLOAD_ASM_DATA(CHECK_JSON_SCHEMA_GENOME.out.gca, cache_dir)
+    // Main data input to this subworkflow is genomic_dataset tuple
+    main:        
+        // Verify genome.json schema
+        CHECK_JSON_SCHEMA_GENOME(genomic_dataset)
+
+        // Download genome data files. Files may or may not include gene models (gff) & or peptides
+        DOWNLOAD_ASM_DATA(CHECK_JSON_SCHEMA_GENOME.out.verified_json, cache_dir)
 
-        if (DOWNLOAD_ASM_DATA.out.gene_gff && DOWNLOAD_ASM_DATA.out.protein_fa) {
-            UNPACK_FILE(DOWNLOAD_ASM_DATA.out.gene_gff, 'gff', DOWNLOAD_ASM_DATA.out.gca)
+        // Check for presence of annotation related output files, then process gene models data if present
+        if (DOWNLOAD_ASM_DATA.out.opt_set) {
 
-            PROCESS_GFF3(UNPACK_FILE.out.uncompressed_file, CHECK_JSON_SCHEMA_GENOME.out.verified_json, UNPACK_FILE.out.gca)
+            // Decompress gff3 file, output with accession in tuple
+            unpacked_gff = UNPACK_GFF3(DOWNLOAD_ASM_DATA.out.opt_set, 'gff')
+            
+            // PROCESS_GFF3(UNPACK_GFF3.out.uncompressed_file, CHECK_JSON_SCHEMA_GENOME.out.verified_json)
+            PROCESS_GFF3(unpacked_gff, CHECK_JSON_SCHEMA_GENOME.out.verified_json)
 
-            CHECK_JSON_SCHEMA_FUNCT(PROCESS_GFF3.out.functional_annotation, PROCESS_GFF3.out.gca)
+            // Verify functional_annotation.json schema
+            CHECK_JSON_SCHEMA_FUNCT(PROCESS_GFF3.out.functional_annotation)
 
-            GFF3_VALIDATION(PROCESS_GFF3.out.gene_models)
+            // Tidy and validate gff3 using gff3validator (NOTE: Requires `module load libffi-3.3-gcc-9.3.0-cgokng6`)
```

Comment:
Whilst the `module` information is useful, I have the feeling this should be in the documentation of the pipeline instead of here...

Reply:
Do we even require to load this module if we are using a container already for genometools ?

Reply:
@JAlvarezJarreta  I agree, but for now its useful until we have proper documentation on this pipeline. Can we stick a pin in this. 

@Dishalodha I've tried without the module loaded and it fails with '_gt: error while loading shared libraries: libffi.so.7_'

Reply:
Module load no longer needed, following bug fix to nextflow.config for genome_prepare